Add documentation for ISV contributors (#345)
red-hat-data-services/odh-deployer: CONTRIBUTING.md

  - The [OWNERS_ALIASES](https://github.com/red-hat-data-services/odh-deployer/blob/main/OWNERS_ALIASES) file of the repository has a list of the people who can review, approve, and qe-approve PRs.

## Testing the PR

- Test the changes locally, by manually running the [deploy.sh](deploy.sh) script from the terminal. This definitely helps in that initial rapid iteration phase.
- Create a RHODS-live image based on the changes made. (See steps on how to [build a RHODS image using rhods-live-builder](https://gitlab.cee.redhat.com/data-hub/rhods-live-builder))
- Use the image to install RHODS operator on an OpenShift cluster (See steps on how to [install a RHODS instance on an OpenShift cluster](https://gitlab.cee.redhat.com/data-hub/olminstall))
- Test the changes over this installed RHODS operator to see if they work as expected.

## ISV Contribution

ISV partners can contribute in this repo to display documentation in RHODS
### Resource definition

In order to contribute to RHODS, we currently support three different object definitions to display different data:

> - [OdhApplication](https://github.com/red-hat-data-services/odh-deployer/blob/main/odh-dashboard/crds/odh-application-crd.yaml): Describe the Application Tile displayed in the ISV section.

Reviewer: ```suggestion
- [OdhApplication](https://github.com/red-hat-data-services/odh-deployer/blob/main/odh-dashboard/crds/odh-application-crd.yaml): Describe the Application Title displayed in the ISV section.
```

- [OdhDocument](https://github.com/red-hat-data-services/odh-deployer/blob/main/odh-dashboard/crds/odh-document-crd.yaml): Represent documentation such as tutorials.
- [OdhQuickStart](https://github.com/red-hat-data-services/odh-deployer/blob/main/odh-dashboard/crds/odh-quick-start-crd.yaml): Quickstart definition.

### Deployment

In order to contribute, just add the desired changes in the documentation and raise a new PR for our team to review it.

> The docs are held in the following path, `/odh-dashboard/crds`, in one of these two folders:

Reviewer: ```suggestion
The docs are held in the following path, `/odh-dashboard`, in one of these two folders:
```

- `apps-managed-service`: For managed installations
- `apps-on-prem`: For self-managed installations.

The steps to add new information are:

1. Set up this project locally following the steps mentioned in the [how to contribute section](#how-to-contribute-code-to-odh-deployer).
2. Create or modify one of the isv folders inside the path mentioned above.

> 3. If you are creating a new folder, add it to the `kustomize` file of one of the two folders mentioned above.

Reviewer: ```suggestion
3. If you are creating a new folder, add it to `kustomization.yaml` file in one of the two folders mentioned above.
```
